src/vehicle.h

```diff
@@ -858,6 +858,23 @@ class vehicle : public JsonSerializer, public JsonDeserializer
 
     /*@}*/
 
+    /**
+     * If target for the turret is set, it returns the set target and unsets target.
+     * Otherwise finds a target and returns it.
+     * @return A target for the turret or turret's position if no target is found.
+     */
+    tripoint turret_acquire_target( vehicle_part &pt );
+    /**
+     * Fires a loaded turret at the target.
+     * @return Number of shots fired.
+     */
+    int turret_fire( vehicle_part &pt, const tripoint &targ );
+
+    void turret_reload( vehicle_part &pt );
```

Comment:
I could wait with the test for your branch then.
Or even better: you can add a test of this type to your branch yourself, to make sure everything works.

With the current master system, it is not possible to actually fire a turret in a sane way without calling `turret_reload`. I could call `vehicle::gain_moves`, but this would be like testing monster attacks by calling `game::do_turn`.